site/content/user-guide/java_client_sender.md

```diff
@@ -0,0 +1,268 @@
++++
+title = "Java - integrate hono sender"
+weight = 280
++++
+
+Simple example of a java main program that sends telemetry data to the default tenant from the registered default device. 
+It shall serve as a blueprint to integrate your existing java source code with hono. 
+<!--more-->
+
+
+
+The goal of the following description is to extend an arbitrary Java program to send telemetry data ("upstream") to Hono.
+
+For that the following example is kept as simple as possible (in the tradition of a classical "Hello World" implementation).
+This means that we will make use of simple constant definitions, deal with exceptions as rarely as possible and use a few System.out's.
+
+The only exception to pure simplicity is that we will send several messages in sequence to show the necessary programming patterns.
+
+{{% warning %}}
+Note that production ready code likely has to think more about error handling and logging than this simple blueprint. 
+{{% /warning %}}
+
+## Generate start project
+
+To have a common base, we start with a simple maven project that generates a java main application by using the maven generate archetype like that:
+
+    $ mvn archetype:generate -B -DarchetypeGroupId=org.apache.maven.archetypes \
+    -DarchetypeArtifactId=maven-archetype-quickstart -DarchetypeVersion=1.1 \
+    -DgroupId=org.eclipse.hono -DartifactId=honoTelemetryUpstreamSender -Dversion=0.1-SNAPSHOT -Dpackage=org.eclipse.hono
+
+If you wish to use your own java project you can skip this step. The integration of the following steps into your existing source code will be straightforward.
+
+## Configure your project
+
+Next we need to extend the maven configuration :
+ 
+### Enable Java 1.8
+Add to your pom.xml:
+```xml
+      <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.3</version>
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+```
+### Add dependency to hono client
+Add to your pom.xml:
+
+```xml
+<dependency>
+      <groupId>org.eclipse.hono</groupId>
+      <artifactId>hono-client</artifactId>
+      <version>0.5-M4</version>
+</dependency>
+```
+### Copy truststore
+
+For the encrypted communication with hono, please copy the provided truststore from the hono project, e.g. like
```

Comment:
Since by default the Hono server is using the *encrypted* connection, I added a hint in a warning box about what to do if an *unencrypted* connection shall be used.